server/src/test/java/org/elasticsearch/search/aggregations/support/TimeSeriesIndexSearcherTests.java

```diff
@@ -95,6 +108,124 @@ public void testCollectInOrderAcrossSegments() throws IOException, InterruptedEx
         dir.close();
     }
 
+    static class StaticScoreFunctionBuilder extends ScoreFunctionBuilder<StaticScoreFunctionBuilder> {
+        private final ScoreFunction scoreFunction;
+
+        StaticScoreFunctionBuilder(ScoreFunction scoreFunction) {
+            this.scoreFunction = scoreFunction;
+        }
+
+        @Override
+        protected void doWriteTo(StreamOutput out) throws IOException {
+
+        }
+
+        @Override
+        public String getName() {
+            return null;
+        }
+
+        @Override
+        protected void doXContent(XContentBuilder builder, Params params) throws IOException {
+
+        }
+
+        @Override
+        protected boolean doEquals(StaticScoreFunctionBuilder functionBuilder) {
+            return false;
+        }
+
+        @Override
+        protected int doHashCode() {
+            return 0;
+        }
+
+        @Override
+        protected ScoreFunction doToFunction(SearchExecutionContext context) throws IOException {
+            return scoreFunction;
+        }
+
+        @Override
+        public TransportVersion getMinimalSupportedVersion() {
+            return null;
+        }
+    }
+
+    public void testCollectMinScoreAcrossSegments() throws IOException, InterruptedException {
```

Comment:
I think this test can be simplified by just using Lucene queries? And in particular the boost query with match all query, which allows for easily checking the minimum score feature works.

Something like this:

```java
public void testCollectMinScoreAcrossSegments() throws IOException {
        Directory dir = newDirectory();
        RandomIndexWriter iw = getIndexWriter(dir);

        AtomicInteger clock = new AtomicInteger(0);

        final int DOC_COUNTS = 5;
        Document doc = new Document();
        for (int j = 0; j < DOC_COUNTS; j++) {
            String tsid = "tsid" + j % 30;
            long time = clock.addAndGet(j % 10);
            doc.clear();
            doc.add(new SortedDocValuesField(TimeSeriesIdFieldMapper.NAME, new BytesRef(tsid)));
            doc.add(new NumericDocValuesField(DataStream.TimestampField.FIXED_TIMESTAMP_FIELD, time));
            try {
                iw.addDocument(doc);
            } catch (IOException e) {
                throw new UncheckedIOException(e);
            }
        }
        iw.close();

        IndexReader reader = DirectoryReader.open(dir);
        IndexSearcher searcher = new IndexSearcher(reader);

        TimeSeriesIndexSearcher indexSearcher = new TimeSeriesIndexSearcher(searcher, List.of());
        indexSearcher.setMinimumScore(2f);

        {
            var collector = new TimeSeriesCancellationTests.CountingBucketCollector();
            var query = new BoostQuery(new MatchAllDocsQuery(), 2f);
            indexSearcher.search(query, collector);
            assertThat(collector.count.get(), equalTo(DOC_COUNTS));
        }
        {
            var collector = new TimeSeriesCancellationTests.CountingBucketCollector();
            var query = new BoostQuery(new MatchAllDocsQuery(), 1f);
            indexSearcher.search(query, collector);
            assertThat(collector.count.get(), equalTo(0));
        }

        reader.close();
        dir.close();
    }
```